ldk/javascript/src/clipboard/index.ts

```diff
@@ -20,21 +20,60 @@ export interface Clipboard {
   /**
    * Starts listening to changes to the clipboard.
    *
-   * @param includeOliveHelpsEvents - if passed in true, callback will be called while olive helps window is in focus
+   * @param filterOptions - object, includesOliveHelps will callback will be called while olive helps window is in focus if true
    * @param callback - A function that's called whenever the clipboard's contents change.
    */
   listen(
-    includeOliveHelpsEvents: boolean,
+    filterOptions: FilterOptions,
     callback: (clipboardText: string) => void,
   ): Promise<Cancellable>;
 }
 
+type FilterOptions = {
+  includeOliveHelpsEvents: boolean;
+};
+
 export function listen(
-  includeOliveHelpsEvents: boolean,
+  filterOptions: FilterOptions,
   callback: (clipboardText: string) => void,
 ): Promise<Cancellable> {
-  oliveHelps.clipboard.includeOliveHelpsEvents(includeOliveHelpsEvents);
-  return promisifyListenable(callback, oliveHelps.clipboard.listen);
+  return new Promise((resolve, reject) => {
+    try {
+      let focusedWindow: string;
+
+      const setFocusedWindow = () =>
+        new Promise((res, rej) => {
+          oliveHelps.window.activeWindow((error, event) => {
+            if (error) {
+              return rej(error);
+            }
+            focusedWindow = event.title;
+
+            return res(focusedWindow);
```

Comment:
since we no longer are setting focusedWindow in multiple places i think it might make sense to remove line 50 & just do `return resolve(event.title)`

Then in line 61 do this

`const focusedWindow = await setFocusedWIndow()`

This way we dont have to deal with a global variable

Reply:
done